x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -315,19 +321,20 @@ async function installDataStreamComponentTemplates(params: {
   await Promise.all(
     templateEntries.map(async ([name, body]) => {
       if (isUserSettingsTemplate(name)) {
-        // look for existing user_settings template
-        const result = await retryTransientEsErrors(
-          () => esClient.cluster.getComponentTemplate({ name }, { ignore: [404] }),
-          { logger }
-        );
-        const hasUserSettingsTemplate = result.component_templates?.length === 1;
-        if (!hasUserSettingsTemplate) {
-          // only add if one isn't already present
+        try {
+          // Attempt to create custom component templates, ignore if they already exist
           const { clusterPromise } = putComponentTemplate(esClient, logger, {
             body,
             name,
+            create: true,
           });
-          return clusterPromise;
+          return await clusterPromise;
+        } catch (e) {
+          if (e?.statusCode === 400 && e.body?.error?.reason.includes('already exists')) {
```

Comment:
This is the error message that will be returned by ES if the custom component template already exists. Unfortunately there's not a better machine readable field name.